bin/PullAndProcessGsAnalysisData.sh

```diff
@@ -35,34 +35,30 @@ REAL_USER="$(logname 2>/dev/null || echo 'no login name')"
 
 
 function sanityChecking(){ 
-
+	#
+	local _batch="${1}"
+	local _controlFileBase="${2}"
+	local _controlFileBaseForFunction="${_controlFileBase}.${FUNCNAME[0]}"
+	local _gsSampleSheet
+	#
+	log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Processing batch ${_batch}..."
+	#
+	# Check if function previously finished successfully for this data.
+	#
+	if [[ -e "${_controlFileBaseForFunction}.finished" ]]
+	then
+		log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "${_controlFileBaseForFunction}.finished is present -> Skipping."
+		log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "${FUNCNAME[0]} ${_batch}. OK"
+		return
+	else
+		log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "${_controlFileBaseForFunction}.finished not present -> Continue..."
+		printf '' > "${_controlFileBaseForFunction}.started"
+	fi
+	
 	_numberOfSamplesheets=$(find "${TMP_ROOT_DIR}/tmp/${_batch}/" -maxdepth 1 -mindepth 1 -name 'CSV_UMCG_*.'"${SAMPLESHEET_EXT}" 2>/dev/null | wc -l)
 	if [[ "${_numberOfSamplesheets}" -eq 1 ]]
 	then
 		log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Found: one ${TMP_ROOT_DIR}/tmp/${_batch}/CSV_UMCG_*.${SAMPLESHEET_EXT} samplesheet."
-		local _gsSampleSheet
-		_gsSampleSheet=$(ls -1 "${TMP_ROOT_DIR}/tmp/${_batch}/CSV_UMCG_"*".${SAMPLESHEET_EXT}")
-		#
-		# Make sure:
-		#  1. The last line ends with a line end character.
-		#  2. We have the right line end character: convert any carriage return (\r) to newline (\n).
-		#  3. We remove empty lines: lines containing only white space and/or field separators are considered empty too.
-		#
-		cp "${_gsSampleSheet}"{,.converted} \
-			2>> "${_controlFileBaseForFunction}.started" \
-			&& printf '\n' \
-			2>> "${_controlFileBaseForFunction}.started" \
-			>> "${_gsSampleSheet}.converted" \
-			&& sed -i 's/\r/\n/g' "${_gsSampleSheet}.converted" \
-			2>> "${_controlFileBaseForFunction}.started" \
-			&& sed -i "/^[\s${SAMPLESHEET_SEP}]*$/d" "${_gsSampleSheet}.converted" \
-			2>> "${_controlFileBaseForFunction}.started" \
-		|| {
-			log4Bash 'ERROR' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Failed to convert line end characters and/or remove empty lines for ${_gsSampleSheet}."
-			mv "${_controlFileBaseForFunction}."{started,failed}
-			return
-		}
-
```

Comment:
Why is this sanity checking block removed? Without it the code will fail if the samplesheet contains the frequent "usual formatting mistakes"...

Reply:
The logic behind is, that we don't use the .converted file in the Analysis, but I will put it back

Reply:
If we do not use the "converted" file, the analysis may fail. Simply moving "${_gsSampleSheet}.converted" -> "${_gsSampleSheet}", will cause issues if rsync puts the not-converted version back before you are done with the analysis. So we need a solution to prevent the usual errors from crashing the analysis...